torch/_inductor/fx_passes/b2b_gemm.py

```diff
@@ -154,118 +298,339 @@ def cdiv(x: int, y: int):
         "BLOCK_SIZE_N": 64,
         "BLOCK_SIZE_O": 128,
         "BLOCK_SIZE_P": 64,
+        "num_stages": 2,
+        "num_warps": 4,
+    },
+    {
+        "BLOCK_SIZE_M": 32,
+        "BLOCK_SIZE_N": 64,
+        "BLOCK_SIZE_O": 32,
+        "BLOCK_SIZE_P": 64,
         "num_stages": 4,
         "num_warps": 8,
     },
     {
-        "BLOCK_SIZE_M": 256,
-        "BLOCK_SIZE_N": 32,
-        "BLOCK_SIZE_O": 256,
-        "BLOCK_SIZE_P": 32,
+        "BLOCK_SIZE_M": 32,
+        "BLOCK_SIZE_N": 128,
+        "BLOCK_SIZE_O": 32,
+        "BLOCK_SIZE_P": 128,
         "num_stages": 2,
         "num_warps": 4,
     },
     {
-        "BLOCK_SIZE_M": 256,
-        "BLOCK_SIZE_N": 64,
-        "BLOCK_SIZE_O": 256,
-        "BLOCK_SIZE_P": 64,
-        "num_stages": 3,
+        "BLOCK_SIZE_M": 64,
+        "BLOCK_SIZE_N": 128,
+        "BLOCK_SIZE_O": 64,
+        "BLOCK_SIZE_P": 128,
+        "num_stages": 2,
         "num_warps": 4,
     },
 ]
 
 
-def can_apply_b2b_gemm(match: Match) -> bool:
-    if not all("val" in arg.meta for arg in match.args):
+def is_b2b_gemm_good_on(
+    is_left_assoc: bool,
+    A_node: torch.fx.Node,
+    B_node: torch.fx.Node,
+    C_node: torch.fx.Node,
+) -> bool:
+    """
+    checks whether the sizes are good for b2b_gemm
+    """
+    # basic checks
+    if not all(["val" in A_node.meta, "val" in B_node.meta, "val" in C_node.meta]):
         return False
-    mats = [arg.meta["val"] for arg in match.args]
-    if not all(mat.is_cuda for mat in mats):
+    A, B, C = A_node.meta["val"], B_node.meta["val"], C_node.meta["val"]  # torch._subclasses.fake_tensor.FakeTensor
+    if not all([A.is_cuda, B.is_cuda, C.is_cuda]):
         return False
-    if not all(len(mat.shape) == 2 for mat in mats):
+    if not all([len(A.shape) == 2, len(B.shape) == 2, len(C.shape) == 2]):
         return False
-    mat1, mat2, mat3 = mats
-    if not ((mat1.shape[1] == mat2.shape[0]) and (mat2.shape[1] == mat3.shape[0])):
+    if not ((A.shape[1] == B.shape[0]) and (B.shape[1] == C.shape[0])):
         return False
-    M, N = mat1.shape
-    O, P = mat3.shape
+    # size checks: we only dispatch to B2B-GEMM when the average load ratio is > 1
```

Comment:
we should also set limits on sizes of inputs..

Reply:
"Theoretically" the shared mem only limits block sizes because `M, N, O, P` are all divided into blocks. Are there other limits on `M, N, O, P`?